docs/upgrade.md

```diff
@@ -3,5 +3,288 @@ title: Upgrade guide
+        super(scope, id, props);
+
+        const handler = new NodejsFunction(this, 'helloWorldFunction', {
+          runtime: Runtime.NODEJS_20_X,
+          handler: 'handler',
+          entry: 'src/index.ts',
+          bundling: {
+            format: OutputFormat.ESM,
+            minify: true,
+            esbuildArgs: {
+              "--tree-shaking": "true",
+            },
+            banner: 
+              "import { createRequire } from 'module';const require = createRequire(import.meta.url);", // (1)!
+          },
+        });
+      }
+    }
+    ```
+    
+    1. `esbuild` will include this arbitrary code at the top of your bundle to maximize CommonJS compatibility _(`require` keyword)_.
+
+=== "With AWS SAM"
+
+    ```yaml hl_lines="14 17-18"
+    Transform: AWS::Serverless-2016-10-31
+    Resources:
+      HelloWorldFunction:
+        Type: AWS::Serverless::Function
+        Properties:
+          Runtime: nodejs20.x
+          Handler: src/index.handler
+        Metadata:
+          BuildMethod: esbuild
+          BuildProperties:
+            Minify: true
+            Target: 'ES2020'
+            Sourcemap: true
+            Format: esm
+            EntryPoints:
+              - src/index.ts
+            Banner:
+              js: "import { createRequire } from 'module';const require = createRequire(import.meta.url);"  # (1)!
+ 
+    ```
+
+    1. `esbuild` will include this arbitrary code at the top of your bundle to maximize CommonJS compatibility _(`require` keyword)_.
+
+## Scoped imports
+
+### Middy.js middleware imports
+
+???+ note "Disregard if you are not using Middy.js middlewares."
+    In v2, we've added support for subpath exports. This means if you don't import Middy.js middlewares, you will benefit from a smaller bundle size.
+
+In v1, you could import Middy.js middlewares from the default export of a package _(e.g., `logger`)_. For example, you'd import `injectLambdaContext` Logger middleware from `@aws-lambda-powertools/logger`.
+
+In v2, you can now import only the Middy.js middlewares you want to use from a subpath export, _e.g., `@aws-lambda-powertools/logger/middleware`_, leading to a smaller bundle size.
+
+=== "Before"
+
+    ```typescript
+    import { Logger, injectLambdaContext } from '@aws-lambda-powertools/logger';
+    import { Tracer, captureLambdaHandler } from '@aws-lambda-powertools/tracer';
+    import { Metrics, logMetrics } from '@aws-lambda-powertools/metrics';
+    ```
+
+=== "After"
+
+    ```typescript hl_lines="2 4 6"
+    import { Logger } from '@aws-lambda-powertools/logger';
+    import { injectLambdaContext } from '@aws-lambda-powertools/logger/middleware';
+    import { Tracer } from '@aws-lambda-powertools/tracer';
+    import { captureLambdaHandler } from '@aws-lambda-powertools/tracer/middleware';
+    import { Metrics } from '@aws-lambda-powertools/metrics';
+    import { logMetrics } from '@aws-lambda-powertools/metrics/middleware';
+    ```
```

Comment:
```suggestion
    ```typescript hl_lines="2 5 8"
    import { Logger } from '@aws-lambda-powertools/logger';
    import { injectLambdaContext } from '@aws-lambda-powertools/logger/middleware';
    
    import { Tracer } from '@aws-lambda-powertools/tracer';
    import { captureLambdaHandler } from '@aws-lambda-powertools/tracer/middleware';
    
    import { Metrics } from '@aws-lambda-powertools/metrics';
    import { logMetrics } from '@aws-lambda-powertools/metrics/middleware';
    ```
```

Add space to make it easier to grasp before/after. It reads a bit crowded now

<img width="921" alt="image" src="https://github.com/aws-powertools/powertools-lambda-typescript/assets/3340292/f3d5d816-2007-457d-9427-356c6dab1c33">